assets/scripts/app/StreetEditable.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import Segment from '../segments/Segment'
+
+import { TILE_SIZE } from '../segments/constants'
+import { getVariantArray } from '../segments/variant_utils'
+import { generateRandSeed } from '../util/random'
+
+class StreetEditable extends React.Component {
+  static propTypes = {
+    onResized: PropTypes.bool.isRequired,
+    setBuildingWidth: PropTypes.func.isRequired,
+    street: PropTypes.object.isRequired,
+    calculatePerspective: PropTypes.func.isRequired
+  }
+
+  componentDidUpdate (prevProps) {
+    const { onResized } = this.props
+
+    if (onResized && prevProps.onResized !== onResized) {
+      this.props.setBuildingWidth(this.streetSectionEditable)
+    }
+  }
+
+  updateSegmentData = (ref, dataNo, segmentPos) => {
+    const { segments } = this.props.street
+    const segment = segments[dataNo]
+
+    segment.el = ref
+    segment.el.dataNo = dataNo
+    segment.el.savedLeft = Math.round(segmentPos)
+    segment.el.savedNoMoveLeft = Math.round(segmentPos)
+    segment.el.cssTransformLeft = Math.round(segmentPos)
+    segment.el.savedWidth = Math.round(segment.width * TILE_SIZE)
+  }
+
+  calculateSegmentPos = (dataNo) => {
+    const { segments, remainingWidth } = this.props.street
+    let currPos = remainingWidth / 2
+
+    for (let i = 0; i < dataNo; i++) {
+      currPos += segments[i].width
+    }
+
+    return (currPos * TILE_SIZE)
+  }
+
+  renderStreetSegments = () => {
+    const { segments, units } = this.props.street
+    const streetSegments = []
+
+    segments.map((segment, i) => {
```

Comment:
I think we should just `return segments.map()` and then at the end of the anonymous function, return `segmentEl`